tests/storage/test_events_bg_updates.py

```diff
@@ -0,0 +1,156 @@
+                    "update_name": "fixup_max_depth_cap",
+                    "progress_json": "{}",
+                },
+            )
+        )
+
+    def create_room(self, room_version: RoomVersion) -> Dict[str, int]:
+        """Create a room with a known room version and insert events.
+
+        Returns that set of event IDs that exceed MAX_DEPTH.
+        """
+
+        # Create a room with a specific room version
+        self.get_success(
+            self.db_pool.simple_insert(
+                table="rooms",
+                values={
+                    "room_id": self.room_id,
+                    "room_version": room_version.identifier,
+                },
+            )
+        )
+
+        # Insert events with some depths exceeding MAX_DEPTH
+        event_id_to_depth: Dict[str, int] = {}
+        for depth in range(MAX_DEPTH - 5, MAX_DEPTH + 5):
+            event_id = f"$event{depth}:example.com"
+            event_id_to_depth[event_id] = depth
+
+            self.get_success(
+                self.db_pool.simple_insert(
+                    table="events",
+                    values={
+                        "event_id": event_id,
+                        "room_id": self.room_id,
+                        "topological_ordering": depth,
+                        "depth": depth,
+                        "type": "m.test",
+                        "sender": "@user:test",
+                        "processed": True,
+                        "outlier": False,
+                    },
+                )
+            )
+
+        return event_id_to_depth
+
+    def test_fixup_max_depth_cap_bg_update(self) -> None:
+        """Test that the background update correctly caps topological_ordering
+        at MAX_DEPTH."""
+
+        event_id_to_depth = self.create_room(RoomVersions.V6)
+
+        # Run the background update
+        progress = {"room_id": ""}
+        batch_size = 10
+        num_rooms = self.get_success(
+            self.store.fixup_max_depth_cap_bg_update(progress, batch_size)
+        )
+
+        # Verify the number of rooms processed
+        self.assertEqual(num_rooms, 1)
+
+        # Verify that the topological_ordering of events has been capped at
+        # MAX_DEPTH
+        rows = self.get_success(
+            self.db_pool.simple_select_list(
+                table="events",
+                keyvalues={"room_id": self.room_id},
+                retcols=["event_id", "topological_ordering"],
+            )
+        )
+
+        for event_id, topological_ordering in rows:
+            if event_id_to_depth[event_id] >= MAX_DEPTH:
+                # Events with a depth greater than or equal to MAX_DEPTH should
+                # be capped at MAX_DEPTH.
+                self.assertEqual(topological_ordering, MAX_DEPTH)
+            else:
+                # Events with a depth less than MAX_DEPTH should remain
+                # unchanged.
+                self.assertLess(topological_ordering, MAX_DEPTH)
```

Comment:
```suggestion
                self.assertEqual(topological_ordering, event_id_to_depth[event_id])
```

to ensure they haven't been changed.